source/api/content-state/0.9/index.md

```diff
@@ -0,0 +1,713 @@
+
+```json
+{
+  "@context": "http://iiif.io/api/presentation/{{ page.major }}/context.json",
+  "id": "https://example.org/import/1",
+  "type": "Annotation",
+  "motivation": ["contentState"],
+  "target": {
+     "id": "https://example.org/object1/canvas7#xywh=1000,2000,1000,2000",
+     "type": "Canvas",
+     "partOf": [{
+        "id": "https://example.org/object1/manifest",
+        "type": "Manifest"
+     }]
+   }
+}
+```
+
+This description cannot be conveyed by just a Canvas URI or a Manifest URI; it needs the structure provided by a content state Annotation. It can be reduced to _Target body_ form, but no further:
+
+
+```json
+{
+  "id": "https://example.org/object1/canvas7#xywh=1000,2000,1000,2000",
+  "type": "Canvas",
+  "partOf": [{
+    "id": "https://example.org/object1/manifest",
+    "type": "Manifest"
+  }]
+}
+```
+
+
+### 2.3 Protocol and encoding requirements
+
+There are many ways in which the content state data shown in Section 2.2 could be passed to, or exported from, IIIF compatible software such as a viewer. For interoperability, agreeing on the model is not enough. This section defines _Protocols_ for the transfer of this data, so that implementing software can receive or pass a content state without specific knowledge of other participating software. These protocols make use of widely supported features of modern browsers:
+
+* Passing a content state as a query string parameter in an HTTP GET request
+* Passing a content state as a parameter in an HTTP POST request
+* Reacting to the [Paste][org-mozilla-paste] event, where the pasted data is the URI of a content state
+* Reacting to the [Paste][org-mozilla-paste] event, where the pasted data is a full content state
+* Using the [Drag and Drop API][org-mozilla-drag-drop] to expose and accept content states
+* Uploading content state from the client machine via the [FileReader][org-mozilla-filereader] interface
+* Initialising a client via an HTML5 `data-*` attribute
+
+Examples of these protocols are given in Section 3, [Initialization mechanisms][contentstate-initialization].
+
+
+
+#### 2.3.1 Base 64 Encoding with URL and Filename Safe Alphabet
+
+Any content state that is in JSON-LD form, rather than a simple URI string, _MUST_ be encoded as [Base 64 Encoding with URL and Filename Safe Alphabet][org-rfc-4648-5] ("base64url") when declared inline in an HTML document or passed as an HTTP request parameter (e.g., GET or POST), and a client _MUST_ accept it in that form. Note that "base64url" is not the same encoding as "base64", and is used to ensure the integrity of content states passed between web applications.
+
+The destination character set when encoding _MUST_ be UTF-8. For example, when passing the content state as JSON-LD in a query string parameter, base64url is used, as unencoded JSON is vulnerable to corruption. Simple URI forms _SHOULD_ be plain strings.
```

Comment:
Do we need to be explicit about what to do with IRIs? e.g. https://en.wiktionary.org/wiki/Ῥόδος
(e.g. encode directly, or transform into %encoded URI form first)?

Reply:
yes we do...